draco/core/containers.py

```diff
@@ -717,7 +719,95 @@ def is_stacked(self):
         return len(self.stack) != len(self.prod)
 
 
-class Map(ContainerBase):
+class FreqContainer(ContainerBase):
+    """A pipeline container for data with a frequency axis.
+
+    This works like a normal :class:`ContainerBase` container, but already has a freq
+    axis defined, and specific properties for dealing with frequencies.
+    """
+
+    _axes = ("freq",)
+
+    @property
+    def freq(self):
+        """The physical frequency associated with each entry of the time axis.
+
+        By convention this property should return the frequency in MHz at the centre
+        of each of frequency channel.
+        """
+        try:
+            return self.index_map["freq"][:]["centre"]
+        # Need to check for both types as different numpy versions return
+        # different exceptions.
+        except (IndexError, ValueError):
```

Comment:
Nice. can't count the number of times a missing `"centre"` has crashed my code...